test/e2e/subscription_e2e_test.go

```diff
@@ -842,6 +842,124 @@ func TestSubscriptionUpdatesMultipleIntermediates(t *testing.T) {
 	// TODO: check installplans, subscription status, etc
 }
 
+// TestSubscriptionUpdatesExistingInstallPlan ensures that an existing InstallPlan
+//  has the appropriate approval requirement from Subscription.
+func TestSubscriptionUpdatesExistingInstallPlan(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	// Create CSV
+	packageName := genName("nginx-")
+	stableChannel := "stable"
+
+	namedStrategy := newNginxInstallStrategy(genName("dep-"), nil, nil)
+	csvA := newCSV("nginx-a", testNamespace, "", semver.MustParse("0.1.0"), nil, nil, namedStrategy)
+	csvB := newCSV("nginx-b", testNamespace, "nginx-a", semver.MustParse("0.2.0"), nil, nil, namedStrategy)
+
+	// Create PackageManifests
+	manifests := []registry.PackageManifest{
+		{
+			PackageName: packageName,
+			Channels: []registry.PackageChannel{
+				{Name: stableChannel, CurrentCSVName: csvB.GetName()},
+			},
+			DefaultChannelName: stableChannel,
+		},
+	}
+
+	// Create the CatalogSource with just one version
+	c := newKubeClient(t)
+	crc := newCRClient(t)
+	catalogSourceName := genName("mock-nginx-")
+	_, cleanupCatalogSource := createInternalCatalogSource(t, c, crc, catalogSourceName, testNamespace, manifests, nil, []v1alpha1.ClusterServiceVersion{csvA, csvB})
+	defer cleanupCatalogSource()
+
+	// Attempt to get the catalog source before creating install plan
+	_, err := fetchCatalogSource(t, crc, catalogSourceName, testNamespace, catalogSourceRegistryPodSynced)
+	require.NoError(t, err)
+
+	// Create a subscription to just get an InstallPlan for csvB
+	subscriptionName := genName("sub-nginx-")
+	createSubscriptionForCatalog(t, crc, testNamespace, subscriptionName, catalogSourceName, packageName, stableChannel, csvB.GetName(), v1alpha1.ApprovalAutomatic)
+
+	// Wait for csvB to be installed
+	_, err = awaitCSV(t, crc, testNamespace, csvB.GetName(), csvSucceededChecker)
+	require.NoError(t, err)
+
+	subscription, err := fetchSubscription(t, crc, testNamespace, subscriptionName, subscriptionHasInstallPlanChecker)
+	fetchedInstallPlan, err := fetchInstallPlan(t, crc, subscription.Status.InstallPlanRef.Name, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete))
+
+	// Delete this subscription
```

Comment:
Is this step needed? 

Can't we:

- Create CatalogSource with 1 version
- Subscribe to it with manual approval
- Approve the InstallPlan
- Update the CatalogSource to include an update for the operator
- Wait for an installplan
- Ensure that the update to the CSV didn't roll out
- Approve the installplan
- See the update roll out

we don't need to delete subscriptions or manually create installplans, I don't think

Reply:
Given that this is blocking servicemesh, we can revisit this test (in master) instead.